mm2src/coins/z_coin/z_params/indexeddb.rs

```diff
@@ -0,0 +1,178 @@
+            sapling_output: sapling_output.to_vec(),
+            ticker: DB_NAME.to_string(),
+        };
+
+        params_db.add_item(&params).await?;
+
+        Ok(())
+    }
+
+    /// Check if z_params is already save to storage previously.
+    pub(crate) async fn check_params(&self) -> MmResult<bool, ZcoinStorageError> {
+        let locked_db = self.lock_db().await?;
+        let db_transaction = locked_db.get_inner().transaction().await?;
+        let params_db = db_transaction.table::<ZcashParamsWasmTable>().await?;
+
+        let maybe_param = params_db
+            .cursor_builder()
+            .only("ticker", DB_NAME)?
+            .open_cursor("ticker")
+            .await?
+            .next()
+            .await?;
+
+        Ok(maybe_param.is_some())
+    }
+
+    /// Get z_params from storage.
+    pub(crate) async fn get_params(&self) -> MmResult<(Vec<u8>, Vec<u8>), ZcoinStorageError> {
+        let locked_db = self.lock_db().await?;
+        let db_transaction = locked_db.get_inner().transaction().await?;
+        let params_db = db_transaction.table::<ZcashParamsWasmTable>().await?;
+        let mut maybe_params = params_db
+            .cursor_builder()
+            .only("ticker", DB_NAME)?
+            .open_cursor("ticker")
+            .await?;
+
+        let mut sapling_spend = vec![];
+        let mut sapling_output = vec![];
+
+        while let Some((_, params)) = maybe_params.next().await? {
+            sapling_spend.extend_from_slice(&params.sapling_spend);
+            if params.sapling_spend_id == 0 {
+                sapling_output = params.sapling_output
+            }
+        }
+
+        Ok((sapling_spend, sapling_output))
+    }
+
+    /// Download and save z_params to storage.
+    pub(crate) async fn download_and_save_params(&self) -> MmResult<(Vec<u8>, Vec<u8>), ZcoinStorageError> {
+        let (sapling_spend, sapling_output) = super::download_parameters().await.unwrap();
+        let spends = sapling_spend_to_chunks(&sapling_spend);
+        for (i, spend) in spends.into_iter().enumerate() {
+            if i == 0 {
+                self.save_params(i as u8, &spend, &sapling_output).await?
+            } else {
+                self.save_params(i as u8, &spend, &[]).await?
+            }
+        }
+
+        Ok((sapling_spend, sapling_output))
+    }
+}
+
+/// Since sapling_spend data way is greater than indexeddb max_data(267386880) bytes to save, we need to split
+/// sapling_spend into chunks of 12 and insert to db multiple times with index(sapling_spend_id)
+fn sapling_spend_to_chunks(sapling_spend: &[u8]) -> Vec<Vec<u8>> {
+    // Calculate the target size for each chunk
+    let chunk_size = sapling_spend.len() / TARGET_SPEND_CHUNKS;
+    // Calculate the remainder for cases when the length is not perfectly divisible
+    let remainder = sapling_spend.len() % TARGET_SPEND_CHUNKS;
+    let mut sapling_spend_chunks: Vec<Vec<u8>> = Vec::with_capacity(TARGET_SPEND_CHUNKS);
```

Comment:
please use slices `sapling_spend_chunks: Vec<&[u8]>` instead of creating  new vector for each chunk (return will be `Vec<&[u8]>`)